blockchain/types/account/smart_contract_account.go

```diff
@@ -117,21 +145,43 @@ func (sca *SmartContractAccount) EncodeRLP(w io.Writer) error {
 	return rlp.Encode(w, sca.toSerializable())
 }
 
-func (sca *SmartContractAccount) DecodeRLP(s *rlp.Stream) error {
-	serialized := &smartContractAccountSerializable{
-		newAccountCommonSerializable(),
-		common.Hash{},
-		[]byte{},
-		params.CodeInfo(0),
+func (sca *SmartContractAccount) EncodeRLPExt(w io.Writer) error {
```

Comment:
It seems that this implementation can potentially replace `EncodeRLP`, doesn't it?

Reply:
That's possible, because when `StorageRoot.IsLegacy()`, EncodeRLP == EncodeRLPExt.
Let me investigate on it.

Reply:
I see the potential. But I am not yet comfortable with affecting other parts of the code.

The rationale was to minimize impact by having the two:
- (a) `AccountSerializer{ preserveExtHash: false }`. Most use cases still use this serializer.
  - snap sync, to store the node received from a remote peer. I'm not sure if the AccountSnapshot may contain ExtHash. So I'd like to keep the original behavior just in case.
  - (It appears in [GetAccount API](https://github.com/klaytn/klaytn/blob/v1.10.2/api/api_public_blockchain.go#L153), but EncodeRLP is not used. Only MarshalJSON)
- (b) `AccountSerializer{ preserveExtHash: true }`
  - Used for [State object encoding](https://github.com/blukat29/klaytn/blob/trie-pruning-queue/blockchain/state/state_object.go#L133)

Reply:
Got it. Thank you.